### PR TITLE
fix(overlay): reduce circular dependencies

### LINKED ISSUE #3563: [Bug]: New Overlay.open cause Illegal constructor at lit/reactive-element/development/reactive-element.js:102:5
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

@spectrum-web-components/overlay

### Expected behavior

[Overlay.open](https://opensource.adobe.com/spectrum-web-components/components/overlay/) api should work. Here is the function signature of `open`:
```
Overlay.open(
    (owner: HTMLElement), // the element to open the overlay in reference to, "trigger"
    (interaction: TriggerInteractions), // the type of interaction type that opened the overlay
    (overlayElement: HTMLElement), // the element that will be projected into the overlay, "content"
    (options: OverlayOptions) // options to customize the overlay
);
```

### Actual behavior

```
TypeError
Illegal constructor
    at new ReactiveElement (https://nccx6g.csb.app/node_modules/
lit/reactive-element/development/reactive-element.js:102:5
new LitElement
https://nccx6g.csb.app/node_modules/lit-element/development/lit-element.js:27:5
    at new AbstractOverlay (https://nccx6g.csb.app/node_modules/
spectrum-web-components/overlay/src/AbstractOverlay.dev.js:101:5
    at new _Overlay (https://nccx6g.csb.app/node_modules/
spectrum-web-components/overlay/src/Overlay.dev.js:47:5
    at _Overlay.open (https://nccx6g.csb.app/node_modules/
spectrum-web-components/overlay/src/AbstractOverlay.dev.js:152:21
handleButtonClick
/src/App.tsx:10:18
   7 | const handleButtonClick = async () => {
   8 |   const overlayTriggerElement = document.getElementById("overlay-trigger");
   9 |   const overlayContentElement = document.getElementById("popover-content");
> 10 |   await Overlay.open(
     |                ^
  11 |     overlayTriggerElement!,
  12 |     "click",
  13 |     overlayContentElement!,

HTMLUnknownElement.callCallback
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:4164:14
Object.invokeGuardedCallbackDev
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:4213:16
invokeGuardedCallback
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:4277:31
invokeGuardedCallbackAndCatchFirstError
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:4291:25
executeDispatch
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:9041:3
processDispatchQueueItemsInOrder
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:9073:7
processDispatchQueue
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:9086:5
dispatchEventsForPlugins
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:9097:3
eval
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:9288:12
batchedUpdates$1
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:26140:12
batchedUpdates
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:3991:12
dispatchEventForPluginEventSystem
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:9287:3
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:6465:5
dispatchEvent
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:6457:5
dispatchDiscreteEvent
https://nccx6g.csb.app/node_modules/react-dom/cjs/react-dom.development.js:6430:5
```

### Screenshots

_No response_

### What browsers are you seeing the problem in?

Chrome

### How can we reproduce this issue?

1. Open this sandbox - https://codesandbox.io/s/overlay-open-api-nccx6g
2. Click `Hello CodeSandbox` button
3. Observe the screen or browser console

### Sample code that illustrates the problem

_No response_

### Logs taken while reproducing problem

_No response_

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Remove some circular dependencies from the Overlay class architecture.

## Related issue(s)
- fixes #3563

## How has this been tested?
-   [ ] _Test case 1_
    1. All of the tests pass
    2. Not sure how to test the fail context, however.

## Types of changes
-   [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: packages/overlay/src/loader.ts [hunk 1]
```diff
@@ -10,6 +10,33 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+import type {
```
Comment (Rajdeepc) [inline packages/overlay/src/loader.ts:13]: Do you mind writing some comments on this file just to help everyone understand what we are doing here?

File: packages/overlay/src/AbstractOverlay.ts [hunk 2]
```diff
@@ -18,9 +18,10 @@ import type {
     OverlayOptionsV1,
```
Comment (Rajdeepc) [inline packages/overlay/src/AbstractOverlay.ts:18]: Also some comments on this component would be appreciated

File: packages/overlay/src/loader.ts (hunk 1 above)
Comment (Westbrook) [reply packages/overlay/src/loader.ts:13]: Added.

File: packages/overlay/src/AbstractOverlay.ts (hunk 2 above)
Comment (Westbrook) [reply packages/overlay/src/AbstractOverlay.ts:18]: Added.

**Review by Westbrook — COMMENTED** [review]
@Rajdeepc Not 100% sure what's useful in these comments, but let me know if this clarifies a bit the methods in question. Happy to take questions/points of interest if you'd like something more expansive.

**Comment by jianliao** [comment]
Hi @Westbrook,

Thank you for the quick turnaround and creating a PR to address the `Overlay.open()` issue. Your effort is greatly appreciated.

I've pulled down the changes and ran some local tests, but it appears the error is still occurring on my end. Not sure if there are additional steps I might be missing to fully integrate the changes. Here is the detail step:

1. Sync the PR branch to my local box
2. Run `yarn build`
3. Copy all the files in `packages/overlay` to the [repo](https://github.com/jianliao/swc-overlay-bug)'s node_modules folder for replacement
4. Before run the app, first comment out my hack in `App.tsx`.
5. Run the app - `npm run start`
6. When I click the button, the same error occurred.

image

Would you have any insights or suggestions on how to troubleshoot this further?

Thanks again for your attention to this issue!

**Comment by Westbrook** [comment]
@jianliao I've been testing this in context of our library and some places where the cycle here had been pointed out. I've not gotten a chance to confirm anything in the context of your repro. I'll get that on the list here and get back to you with more info.

**Review by jianliao — APPROVED** [review]
LGTM

**Review by Rajdeepc — APPROVED** [review]

**Pull request merged**